### DISCUSSION TITLE
Is there any way for ORing between two variants?

### DISCUSSION THREAD

**Discussion by phoenisx** [joe-bell/cva#197] [Q&A]
## Context

Let's consider the following variants config:
```js
 {
   variants: {
      intent: {
         primary: '',
         secondary: ''
      },
      disabled: {
         true: '',
         false: ''
      },
      loading: {
         true: 'loading',
         false: ''
      }
   },
}
```

I was trying to look for a `compoundVariant` that is minimal and support both `disabled` and `loading` `true` states in a single `compundVariant` object, as below:

```js
{
  compoundVariants: [
      {
         intent: 'primary',
         disabled: false,
         loading: false, // This compundVariant acts like an `AND` operation
         class: 'text-primary hover:text-primary-dark'
      },
      {
         intent: 'primary',
         or: [{disabled: true}, {loading: true}],
         class: 'text-disabled', // Should not have `:hover` state
      }
   ]
}
```

I know the above `compundVariant` is invalid, but is there any workaround to achieve this, without duplicating the objects and supporting a new class?

**Comment by joe-bell** [comment]
Perhaps a hot take, but I would say it's a lot easier to just copy the classes for each condition – which also keeps it flexible for future change

**Reply by phoenisx** [reply]
Thanks for your excellent work, and I appreciate your response.
Though, I partially agree with the statement above.

Since CSS has become quite advanced these days, managing too many class variants is unnecessary, especially when they can have a common base class.

For the e.g. above, I was thinking to do something as follows:
```html
<script>
variants = cva('', {
  variants: {
      ...
      loading: { true: 'loading', false: '' }
  },
  compoundVariants: [
      ...
      {
         intent: 'primary',
         or: [{disabled: true}, {loading: true}],
         class: 'blocked',
      }
   ]
})
</script>

<style>
  :where(.blocked) { @apply text-gray-100 bg-gray-400 cursor-not-allowed; }

  /** If required we could have specific styles later on */
  :where(.blocked.loading) {
     &::after { content: "Loading..."; }
  }
</style>
```

The only benefit I see in my approach is that we can use already-defined CSS classes as building blocks, instead of introducing new classes for each variant.

Again, even I am not sure what the best approach here is. Trying to achieve this would result in higher complexity in your project (and maintenance), but a simplified CSS.
